src/checkout-buttons/strategies/googlepay/googlepay-checkoutcom-button-strategy.spec.ts

```diff
@@ -0,0 +1,191 @@
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('Validates if strategy has been initialized', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                setTimeout(() => {
+                    strategy.initialize(checkoutButtonOptions);
+                }, 0);
+
+                strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.initialize).toHaveBeenCalledTimes(1);
+            });
+
+            it('fails to initialize the strategy if no container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.UndefinedContainer);
+
+                function initialize() {
+                    strategy.initialize(checkoutButtonOptions);
+                }
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.InvalidContainer);
+
+                return expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+
+        beforeAll(() => {
+            checkoutButtonOptions = getCheckoutButtonOptions(Mode.GooglePayCheckoutcom);
+        });
+
+        it('check if googlepay payment processor deinitialize is called', async () => {
+            await strategy.initialize(checkoutButtonOptions);
+
+            strategy.deinitialize();
+
+            expect(paymentProcessor.deinitialize).toBeCalled();
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
+            await strategy.initialize(checkoutButtonOptions);
+
+            strategy.deinitialize();
+
+            if (checkoutButtonOptions.containerId) {
+                const button = document.getElementById(checkoutButtonOptions.containerId);
+
+                expect(button).toHaveProperty('firstChild', null);
+            }
+
+            container = document.createElement('div');
+            document.body.appendChild(container);
+        });
+
+        it('Validates if strategy is loaded before call deinitialize', async () => {
+            await strategy.deinitialize();
+
+            if (checkoutButtonOptions.containerId) {
+                const button = document.getElementById(checkoutButtonOptions.containerId);
+
+                expect(button).toHaveProperty('firstChild', null);
+            }
+
+            container = document.createElement('div');
+            document.body.appendChild(container);
+        });
+    });
+
+    describe('#handleWalletButtonClick', () => {
+        it('handles wallet button event', async () => {
```

Comment:
what's this test trying to assert?
this line:
```js
expect(paymentProcessor.initialize).toHaveBeenCalled();
```
seems to be executed as part of:
```js
await strategy.initialize(checkoutButtonOptions);
```

so, what's the expected behaviour when `walletButton.click();` ?

Reply:
this test got removed, as another test already covers initialization